File: python/src/patterns/event_sourcing/event_store.py
from typing import Dict, List
from .events import AccountEvent

class EventStore:
    def __init__(self) -> None:
      self._store: Dict[str, List[AccountEvent]] = {}
    def append(self, stream_id: str, event: AccountEvent) -> None:
      self._store.setdefault(stream_id, []).append(event)
    def load(self, stream_id: str) -> list[AccountEvent]:
      return list(self._store.get(stream_id, []))

File: typescript/src/patterns/event-sourcing/EventStore.ts
import type { AccountEvent } from './Events.js';

export class EventStore {
  private store = new Map<string, AccountEvent[]>();
  append(streamId: string, event: AccountEvent) {
    const arr = this.store.get(streamId) ?? [];
    arr.push(event);
    this.store.set(streamId, arr);
  }
  load(streamId: string): AccountEvent[] {
    return [...(this.store.get(streamId) ?? [])];
  }
}
